extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -233,5 +236,191 @@ export class ProjectsController {
 		return treeNode instanceof FolderNode ? utils.trimUri(treeNode.root.uri, treeNode.uri) : '';
 	}
 
+	/**
+	 * Imports a new SQL database project from the existing database,
+	 * prompting the user for a name, file path location and extract target
+	 */
+	public async importNewDatabaseProject(context: any): Promise<void> {
+		let model = <ImportDataModel>{};
+
+		try {
+			let profile = context ? <IConnectionProfile>context.connectionProfile : undefined;
+			//TODO: Prompt for new connection addition and get database information if context information isn't provided.
+			if (profile) {
+				model.serverId = profile.id;
+				model.database = profile.databaseName;
+			}
+
+			// Get project name
+			let newProjName = await this.getProjectName(model.database);
+
+			if (!newProjName) {
+				this.apiWrapper.showErrorMessage(constants.projectNameRequired);
+				return;
+			}
```

Comment:
Since this is inside the try block, we can just throw an Error here and have a single exit point for the function.  Easier to debug that way